crates/wasi-common/cap-std-sync/src/net.rs

```diff
@@ -230,21 +227,76 @@ macro_rules! wasi_stream_write_impl {
                 Ok(val)
             }
             async fn readable(&self) -> Result<(), Error> {
-                let (readable, _writeable) = self.0.is_read_write()?;
+                let (readable, _writeable) = is_read_write(&self.0)?;
                 if readable {
                     Ok(())
                 } else {
                     Err(Error::io())
                 }
             }
             async fn writable(&self) -> Result<(), Error> {
-                let (_readable, writeable) = self.0.is_read_write()?;
+                let (_readable, writeable) = is_read_write(&self.0)?;
                 if writeable {
                     Ok(())
                 } else {
                     Err(Error::io())
                 }
             }
+
+            async fn sock_recv<'a>(
```

Comment:
A more informative error message should be returned when the provided flags are not supported. And a better approach may be to check the flags before entering the function, rather than within the function

当提供的标志不受支持时，应返回更多信息的错误消息，更好的做法可能是在进入函数之前检查标志，而不是在函数内部。